src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -173,36 +174,85 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
+                sendTelemetryEvent(DebuggingTelemetry.clickedRunByLine);
+                void this.appShell.withProgress(
+                    { location: ProgressLocation.Notification, title: DataScience.startingRunByLine() },
+                    async () => {
+                        const editor = this.vscNotebook.activeNotebookEditor;
+                        if (!cell) {
+                            const range = editor?.selections[0];
+                            if (range) {
+                                cell = editor?.document.cellAt(range.start);
+                            }
+                        }
+
+                        if (!cell) {
+                            return;
+                        }
+
+                        if (editor) {
+                            if (await this.checkForIpykernel6(editor.document)) {
+                                this.updateToolbar(true);
+                                this.updateCellToolbar(true);
+                                await this.startDebuggingCell(editor.document, KernelDebugMode.RunByLine, cell);
+                            } else {
+                                void this.installIpykernel6();
+                            }
+                        } else {
+                            void this.appShell.showErrorMessage(DataScience.noNotebookToDebug());
+                        }
+                    }
+                );
+            }),
+
+            this.commandManager.registerCommand(DSCommands.RunByLineContinue, (cell: NotebookCell | undefined) => {
                 const editor = this.vscNotebook.activeNotebookEditor;
-                if (editor) {
-                    if (await this.checkForIpykernel6(editor.document)) {
-                        this.updateToolbar(true);
-                        this.updateCellToolbar(true);
-                        void this.startDebuggingCell(editor.document, KernelDebugMode.RunByLine, cell);
-                    } else {
-                        void this.installIpykernel6(editor.document);
+                if (!cell) {
+                    const range = editor?.selections[0];
+                    if (range) {
+                        cell = editor?.document.cellAt(range.start);
                     }
-                } else {
-                    void this.appShell.showErrorMessage(DataScience.noNotebookToDebug());
                 }
-            }),
 
-            this.commandManager.registerCommand(DSCommands.RunByLineContinue, (cell: NotebookCell) => {
+                if (!cell) {
+                    return;
+                }
+
                 const adapter = this.notebookToDebugAdapter.get(cell.notebook);
                 if (adapter && adapter.debugCellUri?.toString() === cell.document.uri.toString()) {
                     adapter.runByLineContinue();
                 }
             }),
 
+            this.commandManager.registerCommand(DSCommands.RunByLineStop, () => {
```

Comment:
Question is the cell specificity needed here? For any editor there would only be one RBL session. So do we need to look for selection and range? Or could we just stop RBL based on the editor?

Reply:
good point